### PR TITLE
Feat/zoom client

### PR DESCRIPTION
**Pull request by DaniilVolobuiev**
feat: video calls. Client receives appointment info before 5 minutes and can start video call.

### CODE REVIEW

File: src/components/Dashboard/AppointmentElement/index.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+const AppointmentElement = ({
+  index,
+  remoteDoctor,
+  patient,
+}: AppointmentElementProps) => {
+  const [show, setShow] = React.useState<boolean>(false);
+
+  const { t } = useTranslation();
+
+  const fullText = patient.notes[0].note;
```
Comment (lyudad) [inline src/components/Dashboard/AppointmentElement/index.tsx]: 1) what if you will not have patient.notes[0]

File: src/components/Dashboard/AppointmentElement/index.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  const cuttedText = patient.notes[0].note.slice(0, 100).trim();
+
+  const getPatientAge = () => {
+    const birthdate = new Date(patient.birthDate);
+    const today = new Date();
+    const diffTime = Number(today) - Number(birthdate);
+    return Math.floor(diffTime / (1000 * 60 * 60 * 24 * 365));
```
Comment (lyudad) [inline src/components/Dashboard/AppointmentElement/index.tsx]: a lot of magic numbers

File: src/components/Dashboard/AppointmentElement/index.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  patient,
+}: AppointmentElementProps) => {
+  const [show, setShow] = React.useState<boolean>(false);
+
+  const { t } = useTranslation();
+
+  const fullText = patient.notes[0].note;
+  const cuttedText = patient.notes[0].note.slice(0, 100).trim();
+
+  const getPatientAge = () => {
```
Comment (lyudad) [inline src/components/Dashboard/AppointmentElement/index.tsx]: 1) where is return type?
2) this function look quet generic why it not in separate file/folder in the root?

File: src/components/Zoom/CallPanel/index.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  isParticipantFullScreen,
+  setIsParticipantFullScreen,
+  setIsSelfFullScreen,
+  setStatus,
+  startVideoButton,
+  startAudioButton,
+  audioStarted,
+  isMuted,
+  videoStarted,
+}: CallPanelProps) => {
```
Comment (lyudad) [inline src/components/Zoom/CallPanel/index.tsx:43]: here should be empty line

File: src/components/Zoom/CallPanel/index.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  const leaveSession = async () => {
+    try {
+      startVideoButton();
+      await client.leave().then(() => {
+        setIsParticipantFullScreen(false);
+        setIsSelfFullScreen(false);
+        setStatus(false);
+      });
+    } catch (error) {}
+  };
```
Comment (lyudad) [inline src/components/Zoom/CallPanel/index.tsx:54]: here should be empty line

File: src/components/Zoom/styles.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
+import styled from 'styled-components';
+import { Participant } from '@zoom/videosdk';
+import { COLOR_THREE, BLACK_BUT_NOT } from '@constants/colors';
+import { AZURE } from './../../constants/colors';
```
Comment (lyudad) [inline src/components/Zoom/styles.ts]: what is it?

File: src/components/Zoom/utils/useActiveShareChange.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import { RefObject } from 'react';
+
+const useActiveShareChange = (
+  client: any,
```
Comment (lyudad) [inline src/components/Zoom/utils/useActiveShareChange.ts]: what is it?

Comment (DaniilVolobuiev) [reply src/components/Zoom/utils/useActiveShareChange.ts]: This is Zoom SDK property. I didnt find the way how apply type for it. P.S I figured out how to do it

File: src/components/Dashboard/AppointmentElement/index.tsx (hunk 1 above)
Comment (DaniilVolobuiev) [reply src/components/Dashboard/AppointmentElement/index.tsx]: fixed

File: src/components/Dashboard/AppointmentElement/index.tsx (hunk 2 above)
Comment (DaniilVolobuiev) [reply src/components/Dashboard/AppointmentElement/index.tsx]: fixed

File: src/components/Dashboard/AppointmentElement/index.tsx (hunk 3 above)
Comment (DaniilVolobuiev) [reply src/components/Dashboard/AppointmentElement/index.tsx]: fixed and moved to separate file

File: src/components/Zoom/CallPanel/index.tsx (hunk 4 above)
Comment (DaniilVolobuiev) [reply src/components/Zoom/CallPanel/index.tsx:43]: fixed

File: src/components/Zoom/CallPanel/index.tsx (hunk 5 above)
Comment (DaniilVolobuiev) [reply src/components/Zoom/CallPanel/index.tsx:54]: fixed

File: src/components/Zoom/styles.ts (hunk 6 above)
Comment (DaniilVolobuiev) [reply src/components/Zoom/styles.ts]: fixed

File: src/components/Dashboard/AppointmentElement/index.tsx (hunk 2 above)
Comment (lyudad) [reply src/components/Dashboard/AppointmentElement/index.tsx]: where?

File: src/components/Zoom/CallPanel/index.tsx [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+  setIsParticipantFullScreen,
+  setIsSelfFullScreen,
+  setStatus,
+  startVideoButton,
+  startAudioButton,
+  audioStarted,
+  isMuted,
+  videoStarted,
+}: CallPanelProps) => {
+  //
```
Comment (lyudad) [inline src/components/Zoom/CallPanel/index.tsx:44]: why commet needed?

File: src/components/Zoom/CallPanel/style.ts [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+`;
+
+export const PanelButtonStyle = styled.button<CallPanelWrapperProps>`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  gap: 8px;
+  background: ${BLACK_BUT_NOT};
+  border-radius: 50%;
+
```
Comment (lyudad) [inline src/components/Zoom/CallPanel/style.ts:35]: why here empty line?

File: src/components/Zoom/CallPanel/style.ts [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+export const EndCallButtonStyle = styled.button<CallPanelWrapperProps>`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  gap: 8px;
+  background: ${SOME_COLOR};
+  border-radius: 50%;
+
+  font-family: ${FONT_ROBOTO};
```
Comment (lyudad) [inline src/components/Zoom/CallPanel/style.ts:66]: same

File: src/components/Zoom/text.txt [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+      </Stack>
+
+      <Stack gap="10px">
+        <button onClick={startSelfVideo}>Start</button>
+        <button onClick={() => stopSelfVideo()}>Stop</button>
+      </Stack>
+    </VideoContainer>
+  );
+
+  const videoRef = React.useRef<HTMLVideoElement>(null);
```
Comment (lyudad) [inline src/components/Zoom/text.txt:33]: why it inthe middle of the component?

File: src/components/Zoom/text.txt [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+  return await getSignature({
+    tpc: 'something',
+    role_type: 1,
+    user_identity: 'doctor_id',
+    session_key: '123',
+  }).then((res) => {
+    return res.error.data;
+  });
+};
+client.init('en-US', 'CDN');
```
Comment (lyudad) [inline src/components/Zoom/text.txt:54]: ? why its strings

File: src/components/Zoom/text.txt [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+const initAndJoinSession = async () => {
+  console.log(client.getCurrentUserInfo());
```
Comment (lyudad) [inline src/components/Zoom/text.txt:56]: what is it?

**Review by lyudad — APPROVED** [review]

**Pull request merged**